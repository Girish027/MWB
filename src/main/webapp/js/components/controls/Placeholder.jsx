import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Placeholder extends PureComponent {
  render() {
    const {
      message, styleOverride, children, ...otherProps
    } = this.props;
    if (children) {
      return (
        <div
          className="vertical-center horizontal-center"
          style={{
            paddingBottom: '20px',
            ...styleOverride,
          }}
          {...otherProps}
        >
          {children}
        </div>
      );
    }
    return (
      <div
        className="vertical-center horizontal-center message-default"
        style={styleOverride}
        {...otherProps}
      >
        {message}
      </div>
    );
  }
}

Placeholder.defaultProps = {
  styleOverride: {},
  message: '',
};

Placeholder.propTypes = {
  children: PropTypes.node,
  styleOverride: PropTypes.object,
  message: PropTypes.string,
};

export default Placeholder;
