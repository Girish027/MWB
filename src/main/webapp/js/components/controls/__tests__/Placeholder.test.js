import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Placeholder from 'components/controls/Placeholder';

describe('<Placeholder />', () => {
  let wrapper;
  let props = {
    message: 'Default message to be shown',
  };

  describe('Creating an instance:', () => {
    test('should exist', () => {
      wrapper = shallow(<Placeholder {...props} />);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots:', () => {
    test('should render correctly when message is provided', () => {
      wrapper = shallow(<Placeholder {...props} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('should render correctly when children is provided', () => {
      wrapper = shallow(<Placeholder>
        {' '}
        <p> hello </p>
        {' '}
      </Placeholder>);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('should render correctly when message and children are not provided', () => {
      wrapper = shallow(<Placeholder />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
