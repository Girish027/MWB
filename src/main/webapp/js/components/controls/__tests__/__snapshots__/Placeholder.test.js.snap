// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Placeholder /> Snapshots: should render correctly when children is provided 1`] = `
<div
  className="vertical-center horizontal-center"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
   
  <p>
     hello 
  </p>
   
</div>
`;

exports[`<Placeholder /> Snapshots: should render correctly when message and children are not provided 1`] = `
<div
  className="vertical-center horizontal-center message-default"
  style={Object {}}
/>
`;

exports[`<Placeholder /> Snapshots: should render correctly when message is provided 1`] = `
<div
  className="vertical-center horizontal-center message-default"
  style={Object {}}
>
  Default message to be shown
</div>
`;
